components/result/run-history_test.go

```diff
@@ -0,0 +1,413 @@
+
+			// Checking the chaos operator running status
+			By("[Status]: Checking chaos operator running status")
+			err = pkg.OperatorStatusCheck(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Operator status check failed, due to {%v}", err)
+
+			// Get Run History
+			runHistory := pkg.GetRunHistoryStatus(&testsDetails, clients)
+
+			// Prepare Chaos Execution
+			By("[Prepare]: Prepare Chaos Execution")
+			err = pkg.PrepareChaos(&testsDetails, &chaosExperiment, &chaosEngine, clients, false)
+			Expect(err).To(BeNil(), "faile to prepare chaos, due to {%v}", err)
+
+			// Checking runner pod running state
+			By("[Status]: Runner pod running status check")
+			_, err = pkg.RunnerPodStatus(&testsDetails, testsDetails.AppNS, clients)
+			Expect(err).To(BeNil(), "Runner pod status check failed, due to {%v}", err)
+
+			// Chaos pod running status check
+			err = pkg.ChaosPodStatus(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Chaos pod status check failed, due to {%v}", err)
+
+			// Waiting for chaos pod to get completed
+			// And print the logs of the chaos pod
+			By("[Status]: Wait for chaos pod completion and then print logs")
+			err = pkg.ChaosPodLogs(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Failed to get the experiment chaos pod logs, due to {%v}", err)
+
+			//Checking chaosengine verdict
+			By("Checking the Verdict of Chaos Engine")
+			chaosEngineVerdict, err := pkg.GetChaosEngineVerdict(&testsDetails, clients)
+			Expect(err).To(BeNil(), "Fail to get the chaosengine Verdict, due to {%v}", err)
+			Expect(chaosEngineVerdict).To(Equal("Pass"), "ChaosEngine Verdict is not Stopped, due to {%v}", err)
+
+			// Checking the chaosresult run history verdict
+			By("[Verdict]: Checking the chaosresult run history verdict")
+			err = pkg.ChaosRunHistoryVerdict(&testsDetails, clients, runHistory)
+			Expect(err).To(BeNil(), "ChaosResult Run History Check failed, due to {%v}", err)
+		})
+	})
+
+	Context("Run history check for failed run", func() {
+
+		It("Should check for the run history in the chaos result when experiment fails", func() {
+
+			testsDetails := types.TestDetails{}
+			clients := environment.ClientSets{}
+			chaosExperiment := v1alpha1.ChaosExperiment{}
+			chaosEngine := v1alpha1.ChaosEngine{}
+
+			//Getting kubeConfig and Generate ClientSets
+			By("[PreChaos]: Getting kubeconfig and generate clientset")
+			err := clients.GenerateClientSetFromKubeConfig()
+			Expect(err).To(BeNil(), "Unable to Get the kubeconfig, due to {%v}", err)
+
+			//Fetching all the default ENV
+			By("[PreChaos]: Fetching all default ENVs")
+			klog.Infof("[PreReq]: Getting the ENVs for the %v test", testsDetails.ExperimentName)
+			environment.GetENV(&testsDetails, "pod-delete", "run-history-failed1")
+			// environment.GetENV(&testsDetails, "pod-delete", "run-history")
+
+			// Changing app name in testDetails for a failed verdict
+			testsDetails.AppLabel = "abcXYZ"
```

Comment:
@uditgaurav do we have to set any default value? cc @avaakash

Reply:
I manually changed the app name to a random value so that the experiment fails, to check failed history. If there is a better way to fail an experiment, we can use that method.

Reply:
use some function to get 6 char random string then.

Reply:
This is a negative case so he's giving a dummy value this should be alright.